Dev

vinayinvicible opened the pull request.

vinayinvicible committed:
initial commit.

vinayinvicible committed:
removed templatetags. can add later if necessary.

vinayinvicible committed:
middleware changed to ignore cms page responses
public_data field changed from textfield to jsonfield as we are django caching.
removed publish_cms_content function and made it a method of the CmsPage model.
made template nullable for BaseCmsPage so as to be compatible with existing urls. added corresopnding validation in clean method
made cms_urlpatterns and url validations to respect APPEND_SLASH setting
fixed template path issue in tests.
modified get_public_data to enable views that use template context.
using JsonResponse wherever needed.

vinayinvicible committed:
added requirements.txt

vinayinvicible committed:
added setup.py script.

vinayinvicible committed:
renamed base template and template folders to avoid conflicts with other apps.

vinayinvicible committed:
renamed legacy variables.
reloading the page after successful submimssion.

vinayinvicible committed:
fixed issue when request is not passed in cms_response

vinayinvicible committed:
minor bug fix.

amitu commented on cfblog/middleware.py:
```diff
@@ -0,0 +1,38 @@
+
+        # we return the original response
+        # if the response is given by cms_page_index by catching url pattern
+        if not request.resolver_match or request.resolver_match.func is not cms_page_index:
```
Do the opposite of this condition and return response.

Basically like above, where you checked for is_ajax() etc, you should always look for "bad conditions" and return, and then start processing, else you have meaningless nesting. Think of everything that can go wrong, and check and return, and then start processing.

amitu commented on cfblog/middleware.py:
```diff
@@ -0,0 +1,38 @@
+        if not request.resolver_match or request.resolver_match.func is not cms_page_index:
+            try:
+                cms_page = CmsPage.objects.get(url=request.path)
+            except:
```
Catch CmsPage.DoesNotExist, not except.

amitu commented on cfblog/middleware.py:
```diff
@@ -0,0 +1,38 @@
+                pass
+            else:
+                # process only successful and 404 responses
+                if response.status_code == 404:
```
I do not follow this. Lets discuss.

amitu commented on cfblog/models.py:
```diff
@@ -0,0 +1,146 @@
+        self.save()
+
+
+class CmsPage(BaseCmsPage):
```
We won't have this model. There is only one model call Content, which has both page and post. There is no difference between page and post. 

Also since the models.py is already for your stuff, you do not need to put Cms before everything. Table name cfblog_content is better than cfblog_cmscontent.

amitu commented on cfblog/models.py:
```diff
@@ -0,0 +1,146 @@
+    def html(self):
+        return get_public_data(self)
+
+    @stale_cache('cms_page_{id}')
```
This should be stales_cache.

amitu commented on cfblog/models.py:
```diff
@@ -0,0 +1,146 @@
+        super(CmsPage, self).clean()
+
+
+class CmsBlogCategory(models.Model):
```
This is over designed. We have simple non nested category. Will have only one fields title, description and slug. Remove level, is_active, and parent.

amitu commented on cfblog/models.py:
```diff
@@ -0,0 +1,146 @@
+        return self.title
+
+
+class CmsBlogPost(BaseCmsPage):
```
This will become Content.

amitu commented on cfblog/models.py:
```diff
@@ -0,0 +1,146 @@
+    author = models.ForeignKey(settings.AUTH_USER_MODEL, blank=True, null=True)
+    category = models.ForeignKey('CmsBlogCategory')
+    tags = TagField()
+    slug = models.SlugField(_('slug'), max_length=70)
```
This is not needed, lets put the whole relative URL here. Slug is like foo-bar. We want /foo-bar/baz/ here.

amitu commented on cfblog/models.py:
```diff
@@ -0,0 +1,146 @@
+    body = models.TextField(_('body'), null=True, blank=True)
+    tease = models.TextField(_('tease'), blank=True,
+                             help_text=_('Concise text suggested. Does not appear in RSS feed.'))
+    keywords = models.TextField(_('keywords'), blank=True,
```
Keywords will be handled by page data. Remove this.

amitu commented on cfblog/models.py:
```diff
@@ -0,0 +1,146 @@
+                                help_text=_('List of keywords, relevant for SEO'))
+    # TODO: Include S3ImageField
+    # thumbnail = ImageWithThumbsField(upload_to=get_image_path, sizes=(('s', 500, 500), ('r', 450, 600), ('t', 200, 260)), null=True, blank=True)
+    infographic = models.ImageField(upload_to=get_image_path, null=True, blank=True)
```
infographic, pdf, allow_comments, sub_category, related_posts, votes to be removed. Keep notes.

amitu commented on cfblog/models.py:
```diff
@@ -0,0 +1,146 @@
+    def save(self, *args, **kwargs):
+        super(CmsBlogPost, self).save(*args, **kwargs)
+
+    def get_previous_post(self):
```
One category will be hard coded as Page. Exclude content with that category for prev and next methods.

amitu commented on cfblog/response.py:
```diff
@@ -0,0 +1,56 @@
+from .validators import validate_and_get_template
+
+
+def cms_response(template_name,
```
This will be called render. The idea is for this render to be drop in replacement of django's render: https://docs.djangoproject.com/en/1.8/topics/http/shortcuts/#render.

amitu commented on cfblog/settings.py:
```diff
@@ -0,0 +1,25 @@
+        'instead got {}'.format(setting, type(setting_value), type(default_value))
+    )
+
+CMS_AUTH_TEST_FUNC = _get_setting('AUTH_TEST_FUNC', lambda user: user.is_staff or user.is_superuser)
```
This is not how we will handle it. This will be managed by permissions. Django created cfblog | content | Can add/delete/update content permissions, we will simply use that.

amitu commented on cfblog/settings.py:
```diff
@@ -0,0 +1,25 @@
+    )
+
+CMS_AUTH_TEST_FUNC = _get_setting('AUTH_TEST_FUNC', lambda user: user.is_staff or user.is_superuser)
+CMS_BLOG_URL_PREFIX = u'{}'.format(_get_setting('CMS_BLOG_URL_PREFIX', 'articles/').strip('/'))
```
There is no prefix, whole(without the domain name part) URL is available in the content.url field.

amitu commented on cfblog/settings.py:
```diff
@@ -0,0 +1,25 @@
+
+CMS_AUTH_TEST_FUNC = _get_setting('AUTH_TEST_FUNC', lambda user: user.is_staff or user.is_superuser)
+CMS_BLOG_URL_PREFIX = u'{}'.format(_get_setting('CMS_BLOG_URL_PREFIX', 'articles/').strip('/'))
+CMS_TEMPLATE_DIRS = _get_setting('CMS_TEMPLATE_DIRS', settings.TEMPLATE_DIRS)
```
Again, this is not needed, any template is fine. No validation.

amitu commented on cfblog/settings.py:
```diff
@@ -0,0 +1,25 @@
+CMS_BLOG_URL_PREFIX = u'{}'.format(_get_setting('CMS_BLOG_URL_PREFIX', 'articles/').strip('/'))
+CMS_TEMPLATE_DIRS = _get_setting('CMS_TEMPLATE_DIRS', settings.TEMPLATE_DIRS)
+
+PAGE_CACHE_TIMEOUT = _get_setting('PAGE_CACHE_TIMEOUT', 60 * 60)
```
This too I am not sure about. Django has a setting for cache, our app should just use it, in fact since we are using cache, that is already there. cache.get() already honours that setting.

amitu commented on cfblog/templates/cms_templates/snippet.html:
```diff
@@ -0,0 +1,2 @@
+hehehe
```
Lets put these templates inside the example folder.

amitu commented on cfblog/urls.py:
```diff
@@ -0,0 +1,25 @@
+    url(r'^{}/'.format(re.escape(CMS_BLOG_URL_PREFIX)),
+        include(blog_patterns), name='cms_blog_posts_index'),
+
+    url(r'^cms/ajax/save/(?P<save_type>.+)/$',
```
This is the only URL we have. Content will be handled by middleware like flat pages. We are flat pages on steroids, thats all.

amitu commented on cfblog/utils.py:
```diff
@@ -0,0 +1,283 @@
+include_html_re = re.compile(r'^(?:(?:[a-z_0-9])+:)?.+$', re.IGNORECASE)
+namespace_re = re.compile(r'^(?:[a-z_0-9])+$', re.IGNORECASE)
+
+NAMESPACE_DELIMITER = '-'
```
Can we use some other namespace delimiter? - is never a namespace delimiter and will be part of many template names. ":" or "|" is better.

vinayinvicible committed:
squashed post and page model into single content model
added migration to add static page category as first entry in the category model
now middleware only handles 404 pages
renamed the function `cms_response` to `render`
removed unnessecary cms settings
removed catchall and blog urls
verifying authentication using permissions
